v2/src/datadog.ts

```diff
@@ -59,7 +59,12 @@ export class Datadog extends Construct {
   }
 
   public addLambdaFunctions(
-    lambdaFunctions: (lambda.Function | lambdaNodejs.NodejsFunction | lambdaPython.PythonFunction)[],
+    lambdaFunctions: (
+      | lambda.Function
+      | lambdaNodejs.NodejsFunction
+      | lambdaPython.PythonFunction
+      | lambda.SingletonFunction
```

Comment:
Can you put this into a named type and then use that type in all the places you're using the union in this PR?